pkg/raft/rawnode.go

```diff
@@ -438,6 +438,18 @@ func newStorageApplyRespMsg(r *raft, ents []pb.Entry) pb.Message {
 	}
 }
 
+// AckApplied acknowledges that the given entries have been applied. Must be
```

Comment:
nit: they're not necessarily applied, right? The commitment here is that they are applied... by the next call to Ready? By the next `.Step` _or_ `.Ready`? Whatever it is, can you make it precise?

Reply:
This one requires that the entries **are** applied, because their size is used for the `MaxUncommittedEntriesSize` "flow control". There is a second call, `AckApplying`, which can be used to only give a promise to apply.

Updated the comments in the last commit.